mlflow/tracking/client.py

```diff
@@ -661,7 +663,47 @@ def register_prompt(
         # Fetch the prompt-level tags from the registered model
         prompt_tags = registry_client.get_registered_model(name)._tags
 
-        return model_version_to_prompt_version(mv, prompt_tags=prompt_tags)
+        prompt_version = model_version_to_prompt_version(mv, prompt_tags=prompt_tags)
+
+        if experiment_id := MLFLOW_EXPERIMENT_ID.get():
+            self._link_prompt_to_experiment(prompt_version, experiment_id)
+
+        return prompt_version
+
+    def _link_prompt_to_experiment(self, prompt_version: PromptVersion, experiment_id: str) -> None:
+        """
+        Update the experiment IDs for a prompt version.
+        """
+
+        def _link_prompt_to_experiment_async() -> None:
+            try:
+                prompt_info = self.get_prompt(prompt_version.name)
+                existing_ids = prompt_info.tags.get(PROMPT_EXPERIMENT_IDS_TAG_KEY, "")
+                existing_ids = existing_ids.rstrip(",").lstrip(",")
+                exp_ids = [eid.strip() for eid in existing_ids.split(",") if eid.strip()]
+                if experiment_id not in exp_ids:
+                    exp_ids.append(experiment_id)
+                    exp_ids = ",".join(exp_ids)
+                    # Use LIKE to match the experiment ID and experiment ID is auto-incremented
+                    # integer. So add comma before and after the list of experiment IDs to
+                    # avoid false matches (e.g., "1" matches "10").
+                    exp_ids = f",{exp_ids},"
```

Comment:
This is not very beautiful, but I couldn't think of a better way to store and query a list type field without changing the schema of RegisteredModel. EntityAssociation is not really applicable for Prompts since it's managed by model registry store. I'm open to any idea here.

Reply:
Sounds good, added a lock and moved the linkage logic so that it's called only when there's cache miss.